src/storage/store/engine.rs

```diff
@@ -153,7 +157,13 @@ impl ShardEngine {
             "commit insert"
         );
 
-        self.trie.insert(hashes.clone()).unwrap();
+        let db = &*self.shard_store.db;
```

Comment:
We should get the db from self instead of reaching into shard_store. Might be a good idea to ensure no one except the engine needs to hold a reference to the db in their struct. But don't need to fix it in this pr.